kin-sdk/kin-sdk-lib/src/test/kotlin/kin/sdk/internal/queue/PaymentQueueManagerTest.kt

```diff
@@ -0,0 +1,170 @@
+package kin.sdk.internal.queue
+
+import com.nhaarman.mockitokotlin2.*
+import kin.base.KeyPair
+import kin.sdk.internal.data.PendingBalanceUpdater
+import kin.sdk.internal.events.EventsManager
+import kin.sdk.queue.PendingPayment
+import org.hamcrest.MatcherAssert.assertThat
+import org.hamcrest.Matchers.equalTo
+import org.junit.Before
+import org.junit.Test
+import java.math.BigDecimal
+
+class PaymentQueueManagerTest {
+
+    object Constants {
+        const val ONE_MILLI = 1000L
+        const val DELAY_BETWEEN_PAYMENTS_MILLIS = 3000L
+        const val QUEUE_TIMEOUT_MILLIS = 5000L
+        const val MAX_NUM_OF_PAYMENTS = 5
+    }
+
+    private var queueScheduler: FakeQueueScheduler = spy() //TODO use a spy in order to be able to verify method calls on that object, is this fine?
+    private var txTaskQueueManager: TransactionTaskQueueManager = mock()
+    private var pendingBalanceUpdater: PendingBalanceUpdater = mock()
+    private var eventsManager: EventsManager = mock()
+
+    private lateinit var paymentQueueManager: PaymentQueueManager
+    private lateinit var destinationAccount: String
+    private lateinit var sourceAccount: String
+    private val amount: BigDecimal = BigDecimal.TEN
+
+    @Before
+    @Throws(Exception::class)
+    fun setUp() {
+        initPaymentQueueManager()
+    }
+
+    private fun initPaymentQueueManager() {
+        destinationAccount = KeyPair.random().accountId
+        sourceAccount = KeyPair.random().accountId
+//        kinAccount = KinAccountImpl(expectedRandomAccount, FakeBackupRestore(), mock(), mock(), mock())
+        paymentQueueManager = PaymentQueueManagerImpl(txTaskQueueManager, queueScheduler, pendingBalanceUpdater, eventsManager,
+                Constants.DELAY_BETWEEN_PAYMENTS_MILLIS, Constants.QUEUE_TIMEOUT_MILLIS, Constants.MAX_NUM_OF_PAYMENTS)
+    }
+
+    @Test
+    fun `enqueue, list size is increased by one`() {
+        //given
+        val pendingPayment = PendingPaymentImpl(destinationAccount, sourceAccount, amount)
+
+        //when
+        paymentQueueManager.enqueue(pendingPayment)
+
+        //then
+        assertThat(paymentQueueManager.pendingPaymentCount, equalTo(1))
+    }
+
+    @Test
+    fun `enqueue, then wait for delay between payments, after the delay the list will be empty`() {
+        //given
+        val pendingPayment = PendingPaymentImpl(destinationAccount, sourceAccount, amount)
+
+        //when
+        paymentQueueManager.enqueue(pendingPayment)
+        // TODO because we know how much time the task should take then i use sleep for that time plus one milli - is this the correct pattern?
```

Comment:
yes, shorten the one millis as it's 1 sec now...

Reply:
shorten to what? half a second is fine?

Reply:
see above

Reply:
the one milli is kind of epsilon, you can do like what Corey do 0.1 of the delay of payments.